lib/Document.php

```diff
@@ -191,14 +195,57 @@ public function download(?bool $postProcess = true, ?string $encoding = null): s
         return $contents;
     }
 
+    /**
+     * Downloads the document data as a stream.
+     * 
+     * @param resource|string|StreamInterface|null $output Optionnaly copy data stream to the given output.
+     *
+     * @param ?string &$encoding Get the document encoding from the HTTP request.
```

Comment:
rather than passing around a string reference, i'd rather set an attribute on the `Document` class and make it accessible from the caller context